### ISSUE TITLE
Calculate original file size (fixes metadata and decryption progress)

### ISSUE THREAD

**Issue by dschuermann** [open-keychain/open-keychain#754]
see https://github.com/open-keychain/open-keychain/blob/master/OpenKeychain/src/main/java/org/sufficientlysecure/keychain/pgp/PgpDecryptVerify.java#L459

When the file has been compressed by OpenKeychain the calculated value is obviously not correct.
Somehow we should be able to get the uncompressed length without decrypting the whole content. GnuPG somehow gets it:
``gpg --list-packets test.gpg`` reveals the original size (raw data):
```
:literal data packet:
    mode b (62), created 1356362981, name="passwords.txt",
    raw data: 8 bytes
```

* maybe the length is at the end of the file not in the header?

**Comment by Valodim** [comment]
The length of literal data is the length of the packet, minus 1 for the mode byte, minus 1 for the length of the filename, minus length of the filename, minus 4 for the timestamp. It's also possible that the packet is a partial packet, in that case the length is unknown. This data is potentially available in the PGPLiteralData object through the BCPGInputStream, but not exposed (on both accounts).

See the parse_plaintext function in [parse-packet.c:2631](http://git.gnupg.org/cgi-bin/gitweb.cgi?p=gnupg.git;a=blob;f=g10/parse-packet.c;h=62320865c6f8085b91a5ce36538e439134f95fdf;hb=fc17562cc4f8d531ae7f0887cf2a96dcc224b021#l2631)

**Comment by Valodim** [comment]
Fixed in 3efaac2175d5b971e83bf4322d221bcde80157dc, tested in 260ffdf588a0fcd6faef7cc5cf026b9c4bba85be

**Comment by Valodim** [comment]
(forgot the metdata-only method and some comments need updating, reopening because I have no time to fix right now)
